### PR TITLE
feat: add rankings endpoints with data already process

### PR DESCRIPTION
**Pull request by juanmahidalgo**

### CODE REVIEW

File: src/ports/analyticsTotalData/component.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    if (!isValid(network, filters)) {
+      return []
+    }
+
+    const query = getAnalyticsDayDataQuery(filters)
+    const { analyticsDayDatas: fragments } = await subgraph.query<{
+      analyticsDayDatas: AnalyticsDayDataFragment[]
+    }>(query)
+
+    console.log('fragments: ', fragments)
```
Comment (cazala) [inline src/ports/analyticsTotalData/component.ts:29]: oops

File: src/adapters/handlers/analyticsDayData.ts [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
+import { AnalyticsTimeframe } from '../../ports/analyticsDayData/types'
+import { getTimestampFromTimeframe } from '../../ports/analyticsDayData/utils'
 import { AppComponents, Context } from '../../types'
 
 export function createAnalyticsDayDataHandler(
-  components: Pick<AppComponents, 'analyticsDayData'>
-): IHttpServerComponent.IRequestHandler<Context<'/analytics/day'>> {
-  const { analyticsDayData } = components
+  components: Pick<AppComponents, 'analyticsData'>
+): IHttpServerComponent.IRequestHandler<Context<'/analytics/:timeframe'>> {
```
Comment (cazala) [inline src/adapters/handlers/analyticsDayData.ts:9]: don't this handler and the one above have the same route? `/analytics/:timeframe`

Comment (cazala) [reply src/adapters/handlers/analyticsDayData.ts:9]: actually i think this handler `src/adapters/handlers/analyticsDayData.ts` is not used at all, we can remove it no?

File: src/ports/analyticsDayData/utils.ts [hunk 3]
```diff
@@ -11,6 +12,16 @@ export const getAnalyticsDayDataFragment = () => `
   }
 `
 
+export const getAnalyticsTotalDataFragment = () => `
```
Comment (cazala) [inline src/ports/analyticsDayData/utils.ts:15]: This is called _getAnalytics_**Total**_DataFragment_, but it's in the _src/ports/analytics_**Day**_Data/utils.ts_ file, is that correct?

File: src/ports/analyticsTotalData/component.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import { Network, AnalyticsDayDataFilters } from '@dcl/schemas'
+import { ISubgraphComponent } from '@well-known-components/thegraph-component'
+import { AnalyticsDayDataFragment, IAnalyticsDayDataComponent } from './types'
+import { getAnalyticsDayDataQuery } from './utils'
+
+export function createAnalyticsDayDataComponent(options: {
```
[the remark this answers is not in the document]
Comment (juanmahidalgo) [reply src/ports/analyticsTotalData/component.ts]: updated!

File: src/ports/analyticsDayData/utils.ts (hunk 3 above)
Comment (juanmahidalgo) [reply src/ports/analyticsDayData/utils.ts:15]: updated!

File: src/adapters/handlers/analyticsDayData.ts (hunk 2 above)
Comment (juanmahidalgo) [reply src/adapters/handlers/analyticsDayData.ts]: yeah, leftovers 🤦 . Removed!

File: src/ports/analyticsTotalData/component.ts (hunk 1 above)
Comment (juanmahidalgo) [reply src/ports/analyticsTotalData/component.ts]: removed :)

File: src/tests/ports/analyticsDayData.spec.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        filters = {
+          network: Network.MATIC,
+          from: 0,
+        }
+        jest
+          .spyOn(collectionsSubgraph, 'query')
+          .mockResolvedValueOnce({ analytics: graphResponse })
+      })
+
+      it('should fetch the Total data en return it', async () => {
```
Comment (cazala) [inline src/tests/ports/analyticsDayData.spec.ts:49]:
```suggestion
      it('should fetch the Total data and return it', async () => {
```

File: src/tests/ports/analyticsDayData.spec.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        filters = {
+          network: Network.MATIC,
+          from: Date.now(),
+        }
+        jest
+          .spyOn(collectionsSubgraph, 'query')
+          .mockResolvedValueOnce({ analytics: graphResponse })
+      })
+
+      it('should fetch the Total data en return it', async () => {
```
Comment (cazala) [inline src/tests/ports/analyticsDayData.spec.ts:82]: should this be `Day` data instead of `Total`?

Comment (juanmahidalgo) [reply src/tests/ports/analyticsDayData.spec.ts]: indeed! updated

Comment (cazala) [reply src/tests/ports/analyticsDayData.spec.ts]: awesome 👍 feel free to merge :shipit:

**Pull request merged**